### PR TITLE
Pull docmanager download UI into separate plugin so that it can be disabled easily

### PR DESCRIPTION
**Pull request by DianeHu**
## References

Partially addresses #5274, I'll be making additional PRs to pull download UI into separate plugins for other extensions.

## Code changes

Pulled all download UI code in docmanager-extension into a separate plugin so that it can be disabled easily if a user wants to customize a JL environment to not have those features available. Specifically, docmanager-extension adds the download option into the file menu, and command palette, so those two places can now be easily modified to not display download as an option.

## User-facing changes

No user-facing changes with the extension enabled as is. If @jupyterlab/docmanager-extension:download is disabled (the ID for the new plugin), then the file menu and command palette no longer show download as an option:
image
image

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by DianeHu** [comment]
Looks like there was a canceled check, going to close and then reopen the PR to trigger another test run

**Comment by jtpio** [comment]
Thanks!

File: packages/docmanager-extension/src/index.ts [hunk 1]
```diff
@@ -336,13 +336,64 @@ export const pathStatusPlugin: JupyterFrontEndPlugin<void> = {
   }
 };
 
+/**
+ * A plugin providing download commands in the file menu and command palette.
+ */
+export const downloadPlugin: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/docmanager-extension:download',
+  autoStart: true,
+  requires: [ITranslator, ICommandPalette, IMainMenu, IDocumentManager],
```
Comment (jtpio) [inline packages/docmanager-extension/src/index.ts:345]: Should `ICommandPalette` and `IMainMenu` be moved to `optional`?

Comment (DianeHu) [reply packages/docmanager-extension/src/index.ts]: Yup--sorry, missed that, just moved it!

**Review by blink1073 — APPROVED** [review]
Thanks!

**Pull request merged**